install/migrations/update_10.0.x_to_10.1.0/oauth.php

```diff
@@ -0,0 +1,107 @@
+$default_key_sign = DBConnection::getDefaultPrimaryKeySignOption();
+
+if (!$DB->tableExists('glpi_oauth_access_tokens')) {
+    $query = "CREATE TABLE `glpi_oauth_access_tokens` (
+        `identifier` varchar(255) NOT NULL,
+        `client` varchar(255) NOT NULL,
+        `date_expiration` timestamp NOT NULL,
+        `user_identifier` varchar(255) DEFAULT NULL,
+        `scopes` text DEFAULT NULL,
+        PRIMARY KEY (`identifier`),
+        KEY `client` (`client`)
+        ) ENGINE=InnoDB DEFAULT CHARSET={$default_charset} COLLATE={$default_collation} ROW_FORMAT=DYNAMIC";
+    $DB->queryOrDie($query, "10.1.0 add table glpi_oauth_access_tokens");
+}
+
+if (!$DB->tableExists('glpi_oauth_auth_codes')) {
+    $query = "CREATE TABLE `glpi_oauth_auth_codes` (
+        `identifier` varchar(255) NOT NULL,
+        `client` varchar(255) NOT NULL,
+        `date_expiration` timestamp NOT NULL,
+        `user_identifier` varchar(255) DEFAULT NULL,
+        `scopes` text DEFAULT NULL,
+        PRIMARY KEY (`identifier`),
+        KEY `client` (`client`)
+        ) ENGINE=InnoDB DEFAULT CHARSET={$default_charset} COLLATE={$default_collation} ROW_FORMAT=DYNAMIC";
+    $DB->queryOrDie($query, "10.1.0 add table glpi_oauth_auth_codes");
+}
+
+if (!$DB->tableExists('glpi_oauth_refresh_tokens')) {
+    $query = "CREATE TABLE `glpi_oauth_refresh_tokens` (
+        `identifier` varchar(255) NOT NULL,
+        `access_token` varchar(255) NOT NULL,
+        `date_expiration` timestamp NOT NULL,
+        PRIMARY KEY (`identifier`),
+        KEY `access_token` (`access_token`)
+        ) ENGINE=InnoDB DEFAULT CHARSET={$default_charset} COLLATE={$default_collation} ROW_FORMAT=DYNAMIC";
+    $DB->queryOrDie($query, "10.1.0 add table glpi_oauth_refresh_tokens");
+}
+
+if (!$DB->tableExists('glpi_oauthclients')) {
+    $query = "CREATE TABLE `glpi_oauthclients` (
+        `identifier` varchar(255) NOT NULL,
+        `id` int unsigned NOT NULL AUTO_INCREMENT COMMENT 'Internal GLPI ID',
+        `name` varchar(255) NOT NULL DEFAULT '',
+        `comment` text DEFAULT NULL,
+        `secret` varchar(255) NOT NULL,
+        `redirect_uri` varchar(255) NOT NULL,
+        `grants` text NOT NULL,
+        `scopes` text NOT NULL,
+        `is_active` tinyint NOT NULL DEFAULT '1',
+        `is_confidential` tinyint NOT NULL DEFAULT '1',
+        PRIMARY KEY (`identifier`),
+        KEY `id` (`id`),
+        KEY `name` (`name`),
+        KEY `is_active` (`is_active`)
+        ) ENGINE=InnoDB DEFAULT CHARSET={$default_charset} COLLATE={$default_collation} ROW_FORMAT=DYNAMIC";
+    $DB->queryOrDie($query, "10.1.0 add table glpi_oauthclients");
+}
+
+$private_key_path = GLPI_CONFIG_DIR . '/oauth.pem';
+$public_key_path = GLPI_CONFIG_DIR . '/oauth.pub';
+if (!file_exists($private_key_path) && !file_exists($public_key_path)) {
+    $migration->displayMessage("Generating OAuth keys");
+    \Glpi\OAuth\Server::generateKeys();
+}
```

Comment:
If keys are stored on filesystem, what would happen if someone forgot to copy them when reinstalling GLPI on another server?

Anyways, storing them in DB does not seems secure, so we should probably keep them here.

Maybe code should be moved in `Update::doUpdates()`, to ensure that missing keys would be regenerated after migrations (like we do for encryption key).